tests/unit/test_charm.py

```diff
@@ -27,560 +26,528 @@ def read_file(path: str) -> str:
-
-    def set_up_active_status_charm(self, patch_multus_is_ready):
-        self.harness.handle_exec("gnbsim", [], result=0)
-        self.harness.add_storage("config", attach=True)
-        self.harness.set_can_connect(container="gnbsim", val=True)
-        self._n2_data_available()
-        patch_multus_is_ready.return_value = True
-        self.harness.evaluate_status()
-        self.assertEqual(self.harness.charm.unit.status, ActiveStatus())
-
-    def _create_n2_relation(self) -> int:
-        """Create a relation between gnbsim and amf.
-
-        Returns:
-            int: Id of the created relation
-        """
-        amf_relation_id = self.harness.add_relation(relation_name="fiveg-n2", remote_app="amf")
-        self.harness.add_relation_unit(relation_id=amf_relation_id, remote_unit_name="amf/0")
-        return amf_relation_id
-
-    def _n2_data_available(self) -> int:
-        """Create the N2 relation, sets the relation data in the n2 relation and returns its ID.
-
-        Returns:
-            int: ID of the created relation
-        """
-        amf_relation_id = self._create_n2_relation()
-        self.harness.update_relation_data(
-            relation_id=amf_relation_id,
-            app_or_unit="amf",
-            key_values={
-                "amf_hostname": "amf",
-                "amf_port": "38412",
-                "amf_ip_address": "1.1.1.1",
-            },
-        )
-        return amf_relation_id
+    Returns:
+        int: Id of the created relation
+    """
+    amf_relation_id = harness.add_relation(relation_name="fiveg-n2", remote_app="amf")
+    harness.add_relation_unit(relation_id=amf_relation_id, remote_unit_name="amf/0")
+    return amf_relation_id
 
-    def test_given_default_config_when_config_changed_then_status_is_blocked(
-        self,
-    ):
-        self.harness.update_config(key_values={"usim-opc": ""})
-        self.harness.evaluate_status()
+def set_n2_relation_data(harness) -> int:
+    """Create the N2 relation, sets the relation data in the n2 relation and returns its ID.
 
-        self.assertEqual(
-            self.harness.charm.unit.status,
-            BlockedStatus("Configurations are invalid: ['usim-opc']"),
-        )
+    Returns:
+        int: ID of the created relation
+    """
+    amf_relation_id = create_n2_relation(harness)
+    harness.update_relation_data(
+        relation_id=amf_relation_id,
+        app_or_unit="amf",
+        key_values={
+            "amf_hostname": "amf",
+            "amf_port": "38412",
+            "amf_ip_address": "1.1.1.1",
+        },
+    )
+    return amf_relation_id
+
+def set_up_active_status_charm(harness):
+    harness.handle_exec("gnbsim", [], result=0)
+    harness.add_storage("config", attach=True)
+    harness.set_can_connect(container="gnbsim", val=True)
+    set_n2_relation_data(harness)
+    harness.evaluate_status()
+    assert harness.charm.unit.status == ActiveStatus()
+
+
+@pytest.fixture
+def harness(mocker):
```

Comment:
This fixture is defined below a few functions that actually use it. The readability can be improved if the fixture is moved to the top, I think.